austrakka/components/sequence/funcs.py

```diff
@@ -174,15 +179,18 @@ def _get_file(filepath: str) -> SeqFile:
 
 def _post_fastq(sample_files: list[SeqFile], custom_headers):
     files = [file.multipart for file in sample_files]
-    resp = api_post_multipart(
+    resp = api_post_multipart_raw(
         path="/".join([SEQUENCE_PATH, FASTQ_PATH]),
         files=files,
         custom_headers=custom_headers,
     )
 
-    hashes = [FileHash(filename=f.filename, sha256=f.sha256)
-              for f in sample_files]
-    _verify_hash(hashes, resp)
+    data = get_response(resp, True)
+
```

Comment:
Just tracing through the difference now that it uses `api_post_multipart_raw()` followed by `get_response()` - is the only difference that it's setting `log_resp=True` in the `get_response()` call?

Reply:
api_post_multipart() uses httpx.client through decorator injection, which is a form of hardcoding. In that instance, the client is configured to return response.data rather than response. There was no way to examine response.status_code. api_post_multipart_raw() returns the raw response object. After using the response object, it needs to continue with get_response() which has the side-effect of throwing exceptions for non 2xx responses.

Reply:
In direct answer to your question, get_response() here is being called exactly the same way as the injected client and is not an important detail.

Reply:
No, I see that is True in the `@_use_http_client` decorator anyway. I'm not spotting the difference!

Reply:
Ah, that doesn't need to be set when calling the decorator. Is that's what you meant? I think so too, but I don't now if a future version of the decorator would have some other side-effect that becomes relevant. Best to just keep it.